### PR TITLE
[nrf toup] add option to disable nv_counter

### PR DESCRIPTION
**Pull request by hakonfam**
Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

**Comment by hakonfam** [comment]
See NCSDK-9901

**Comment by oyvindronningstad** [comment]
I would prefer to see what upstream says about this before merging it, since I'm not really familiar with NV counters. We're not using "proper" NV counters, but I don't know whether it's still used in regular flash.

**Comment by hakonfam** [comment]
I see your point. I guess the most important item we need to agree on is that building a TF-M image without support for nv_counters makes sense. IMO it does, since we don't need that in SPM today. Also, since I'm able to remove the functionality, and still build a working TF-M secure image I consider it "safe" in the configuration we use.

**Comment by oyvindronningstad** [comment]
I checked now, and it seems it is only provided as an API, not used by other partitions. So I think this is safe enough.

File: trusted-firmware-m/config/config_default.cmake [hunk 1]
```diff
@@ -132,6 +132,7 @@ set(ATTEST_INCLUDE_OPTIONAL_CLAIMS      ON          CACHE BOOL      "Include opt
 set(ATTEST_INCLUDE_COSE_KEY_ID          OFF         CACHE BOOL      "Include COSE key-id in initial attestation token")
 
 set(TFM_PARTITION_PLATFORM              ON          CACHE BOOL      "Enable Platform partition")
+set(PLATFORM_NV_COUNTER_MODULE_DISABLED FALSE       CACHE BOOL      "Disable Non-volatile counter module")
```
Comment (oyvindronningstad) [inline trusted-firmware-m/config/config_default.cmake:135]: I don't personally like the _DISABLED negative polarity. I realize it's convention though. Take it as you will :D

**Review by oyvindronningstad — APPROVED** [review]

File: trusted-firmware-m/config/config_default.cmake (hunk 1 above)
Comment (hakonfam) [reply trusted-firmware-m/config/config_default.cmake:135]: Hehe, I agree. But sticking with convention wins here IMO

**Review by tejlmand — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**